pkg/innerring/internal/blockchain/blockchain.go

```diff
@@ -0,0 +1,398 @@
+	case cfg.P2P.ProtoTickInterval <= 0:
+		panic("non-positive proto tick interval")
+	}
+
+	if opts.Logger == nil {
+		opts.Logger = zap.NewNop()
+	}
+
+	nodeAcc, err := utilConfig.LoadAccount(cfg.Wallet.Path, "", cfg.Wallet.Password)
+	if err != nil {
+		return nil, fmt.Errorf("read node account: %w", err)
+	}
+
+	standByCommittee := make([]string, len(cfg.Committee))
+	for i := range cfg.Committee {
+		standByCommittee[i] = hex.EncodeToString(cfg.Committee[i].Bytes())
+	}
+
+	var cfgBase config.Config
+
+	cfgBaseProto := &cfgBase.ProtocolConfiguration
+	cfgBaseProto.Magic = cfg.NetworkMagic
+	cfgBaseProto.StandbyCommittee = standByCommittee
+	cfgBaseProto.ValidatorsCount = len(standByCommittee)
+	cfgBaseProto.TimePerBlock = cfg.BlockInterval
+	cfgBaseProto.SeedList = opts.SeedNodes
+	cfgBaseProto.VerifyTransactions = true
+	cfgBaseProto.P2PSigExtensions = !opts.NotaryDisabled
+	cfgBaseProto.MaxTraceableBlocks = cfg.TraceableChainLength
+	cfgBaseProto.Hardforks = opts.HardForks
+
+	cfgBaseApp := &cfgBase.ApplicationConfiguration
+	cfgBaseApp.Relay = true
+	cfgBaseApp.UnlockWallet = cfg.Wallet
+	cfgBaseApp.Consensus.Enabled = true
+	cfgBaseApp.Consensus.UnlockWallet = cfg.Wallet
+	cfgBaseApp.P2PNotary.Enabled = !opts.NotaryDisabled
+	cfgBaseApp.P2PNotary.UnlockWallet = cfg.Wallet
+	cfgBaseApp.StateRoot.UnlockWallet = cfg.Wallet
+	cfgBaseApp.RPC.Enabled = true
+	cfgBaseApp.RPC.Addresses = opts.RPCListenAddresses
+	cfgBaseApp.RPC.StartWhenSynchronized = true
+	cfgBaseApp.RPC.MaxGasInvoke = fixedn.Fixed8FromInt64(100)
+	cfgBaseApp.P2P.Addresses = opts.P2PListenAddresses
+	cfgBaseApp.P2P.DialTimeout = cfg.P2P.DialTimeout
+	cfgBaseApp.P2P.ProtoTickInterval = cfg.P2P.ProtoTickInterval
+	cfgBaseApp.P2P.PingInterval = cfg.P2P.Ping.Interval
+	cfgBaseApp.P2P.PingTimeout = cfg.P2P.Ping.Timeout
+	cfgBaseApp.P2P.MinPeers = int(cfg.P2P.MinPeers)
+	cfgBaseApp.P2P.AttemptConnPeers = int(cfg.P2P.AttemptConnPeers)
+	cfgBaseApp.P2P.MaxPeers = int(cfg.P2P.MaxPeers)
+	// TODO: profiler, metrics
+
+	cfgServer, err := network.NewServerConfig(cfgBase)
+	if err != nil {
+		return nil, fmt.Errorf("compose NeoGo server config from the base one: %w", err)
+	}
+
+	var cfgDB dbconfig.DBConfiguration
+	cfgDB.Type = cfg.Storage.typ
+	if cfgDB.Type == dbconfig.BoltDB {
+		cfgDB.BoltDBOptions.FilePath = cfg.Storage.path
+	} else if cfgDB.Type == dbconfig.LevelDB {
+		cfgDB.LevelDBOptions.DataDirectoryPath = cfg.Storage.path
+	}
+
+	blockchainStorage, err := storage.NewStore(cfgDB)
+	if err != nil {
+		return nil, fmt.Errorf("init storage for blockchain: %w", err)
+	}
+
+	blockChain, err := core.NewBlockchain(blockchainStorage, cfgBase.Blockchain(), opts.Logger)
+	if err != nil {
+		closeErr := blockchainStorage.Close()
+		if closeErr != nil {
+			return nil, fmt.Errorf("init core blockchain component: %w; failed to close blockchain storage: %v", err, closeErr)
+		}
+
+		return nil, fmt.Errorf("init core blockchain component: %w", err)
+	}
+
+	go blockChain.Run()
```

Comment:
So the neo-go's blockchain is run inside the neofs-node's blockchain constructor. What happens if neofs'node's blockchain will be created but won't be run (due to some error on initialisation, probably)? Will the neo-go's blockchain be stopped in this case?

Reply:
ur right we should stop it then